vortex-array/src/nbytes.rs

```diff
@@ -1,16 +1,24 @@
+use rust_lapper::{Interval, Lapper};
+
 use crate::{Array, ArrayVisitorExt};
 
 pub trait NBytes: Array {
     /// Total size of the array in bytes, including all children and buffers.
     // TODO(ngates): this should return u64
     fn nbytes(&self) -> usize {
-        let mut nbytes = 0;
+        let mut intervals = Vec::new();
         for array in self.depth_first_traversal() {
             for buffer in array.buffers() {
-                nbytes += buffer.len();
+                let slice: &[u8] = buffer.inner().as_ref();
+                let start = slice.as_ptr() as usize;
+                intervals.push(Interval {
+                    start,
+                    stop: start + slice.len(),
+                    val: true,
+                });
             }
         }
-        nbytes
+        Lapper::new(intervals).cov()
```

Comment:
Are we sure there will not be overlap between buffers at different depths?

Reply:
I wonder if you want to hold all intervals at all level in a single internal tree?

Reply:
there might be overlap between buffers at different depths but I believe we want to count them once as well, maybe